### DISCUSSION TITLE
Building PDF and getting "PLACEHOLDER NEEDED" message...

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#3890] [Q&A] [answered]
I've been merging lots of changes from different people on my book project, and I started getting a PLACEHOLDER NEEDED message when I run Quarto to generate the PDF. I'm wondering where it is coming from. It needs more context to be a useful message, but I suspect it's related to my exercises like:

```markdown
::: {#exr-GRASP_GOF_contrôleur}
```

I found a debug message in https://github.com/quarto-dev/quarto-cli/blob/9aca80aeb66faa07f93a3939453f658ccfd25075/src/resources/filters/crossref/theorems.lua#L54 but I'm not sure if it's what's happening. 

My PDF seems to be OK (all the exercises seem to be there...). Anyway, just some feedback about the message.

I'm seeing it on Windows with 1.3.24, just before the `rsvg-convert` step in my `pdflatex` build:

```
$ quarto preview --to pdf
Terminating existing preview server....DONE

Preparing to preview
...
...
...
  title: Analyse et conception de logiciels
  author:
    - Christopher Fuhrman
    - Yvan Ross
  date: '2023-01-08T17:09:38-05:00'
  date-format: 'DD MMMM YYYY à HH:mm:ss Z'

PLACEHOLDER NEEDED

(rsvg-convert.exe:18084): Pango-WARNING **: couldn't load font "xkcd Script Italic Not-Rotated 14.5595703125", falling back to "Sans Italic Not-Rotated 14.5595703125", expect ugly output.
running pdflatex - 1
  This is pdfTeX, Version 3.141592653-2.6-1.40.24 (MiKTeX 22.12) (preloaded format=pdflatex.fmt)
   restricted \write18 enabled.
  entering extended mode

running pdflatex - 2
  This is pdfTeX, Version 3.141592653-2.6-1.40.24 (MiKTeX 22.12) (preloaded format=pdflatex.fmt)
   restricted \write18 enabled.
  entering extended mode
```

**Comment by cscheid** [comment, marked as answer]
That's a stray debugging print I left out unintentionally, I apologize.

**Reply by fuhrmanator** [reply]
Since I am using the pre-release, debugging prints are OK with me. Cheers!
